intro-dplyr.Rmd

```diff
@@ -0,0 +1,343 @@
+3. [dplyr vignette](https://cran.r-project.org/web/packages/dplyr/vignettes/dplyr.html)
+4. DataCamp course ["Working with Data in the Tidyverse"](https://www.datacamp.com/courses/working-with-data-in-the-tidyverse)
+    - Chapter 1 "Explore your data" is free (the following chapters are available after purchasing the DataCamp subscription)
+
+**For help**:
+    
+1. [Data wrangling cheatsheet](https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf)
+2. [StackOverflow for tidyr](https://stackoverflow.com/questions/tagged/tidyr)
+3. [StackOverflow for dplyr](https://stackoverflow.com/questions/tagged/dplyr)
+4. Quick package information using RStudio: `?tidyr`, `?dplyr`
+
+## Quickly look over the data we will use
+
+```{r}
+library(dplyr)
+library(tidyr)
+```
+
+### Exercise: Choose and look at a dataset
+
+Create an `exercise.R` file (we will have done that in class) and put these code into the file.
+
+```{r}
+# Load the packages
+library(dplyr)
+library(tidyr)
+
+# Variables
+names(CO2)
+
+# General contents
+str(CO2)
+
+# Using the dplyr function
+glimpse(CO2)
+
+# Quick summary
+summary(CO2)
+```
+
+
+## "Messy" vs. "tidy" data
+
+**The dataset is tidy when:**
+
+- Each variable has its own column.
+- Each observation has its own row
+- Each value has its own cell
+
+```{r}
+library(tidyr)
+library(dplyr)
+```
+
+Explore the structure of `table1`. Is it tidy?
+
+```{r}
+table1
+```
+
+**Benefits of tidy data and tidy code**:
+
+1. Time spent preparing your data at the beginning can save hours of frustration in the long run.
+2. "Tidy data" provides a concept and the package tidyr provides the functions for this work.
+3. Tibble package makes datasets efficient to work with.
+4. dplyr provides fast and intuitive data processing functions.
+    - For *very very large* datasets, the data.table package is very fast and powerful
+5. The `%>%` pipe operator can help clarify complex data processing workflows.
+
+## Filtering data based on values and ordering data by variables
+
+```{r}
+who6 <- who5 %>%
+  # exclude relapse of TB, include years after 1997 and non-zero case counts
+  filter(type != "rel", year > 1997, cases > 0) %>%
+  # create a new variable called "nordic country"
+  mutate(
+    nordic = if_else(country == "Denmark" | country == "Sweden" | country == "Norway", "Nordic", "not Nordic")
+  ) %>%
+  # arrange data by year and country
+  arrange(year, nordic)
+```
+
+### Exercise: filter and re-arrange your dataset
+
+Time: 15 min
+
+```{r}
+# make a dataset who7 that would include only smear positive TB diagnosed from 1994 through 2012
+who7 <- who5 %>%
+  filter(type == "sp", year >= 1994) %>%
+  # select all variables except for age
+  select(-age) %>%
+  # arrange by number of cases in descending order
+  arrange(desc(cases))
+```
+
+If you are familiar with ggplot2, then you can optional visualize a bar plot of the case count on y-axis and year on x-axis, with sex as the fill.
```

Comment:
Here I moved the ggplot code out into a text instruction, to show they don't have to do it.